### PR TITLE
feat: add a new command to show the detailed information of measurements

### LINKED ISSUE #640: `show measurements` should return retention policy information
### Is your feature request related to a problem? Please describe.（如果该特性或者功能和已知某个问题相关，请描述问题是什么）

currently, the `show measurement` statement returns measurements from all retention policies, without the retention policy information, this result in two issues:

* it is hard to understand which retention policy a measurement belongs to.
* if there measurement with same name in different retention policies, only one will be listed.

### Describe the solution you'd like（问题的解决办法是什么）

include retention policy information in the output of `show measurements`

### Describe alternatives you've considered（描述您考虑过的替代方案）

_No response_

### Additional context（其他补充内容）

_No response_

### PR DESCRIPTION
**Pull request by StepY1aoZz**
### What problem does this PR solve?
resolve #640 

### What is changed and how it works?
A new command `SHOW MEASUREMENTS DETAIL` was added. Please ref discussion  https://github.com/openGemini/openGemini/discussions/706 for more info.
For now, the return value looks like this:
```
> show measurements detail with measurement = mst
name: mst
tags: Retention policy=rp0
+--------------------------------+
|             Detail             |
+--------------------------------+
| INDEX: <nil>                   |
| SHARD KEY: <nil>               |
| ENGINE TYPE: tsstore           |
| TAG KEYS: country, name        |
| FIELD KEYS: address(string),   |
| age(float), alive(boolean),    |
| height(integer)                |
+--------------------------------+
1 columns, 5 rows in set

``` 

### How Has This Been Tested?

New test case has been added into `server_test.go`.

- [ ] Test A
- [ ] Test B
- [x] Test cases to be added
- [ ] No code

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/util/lifted/influx/coordinator/statement_executor.go [hunk 1]
```diff
@@ -1402,6 +1407,130 @@ func (e *StatementExecutor) executeShowMeasurementsStatement(q *influxql.ShowMea
...
+		return s
+	}
+
+	// sliceFirstColumn gets all shard keys and indexes from the returned models.Row.Values array, then returns a sorted slice.
+	// sk: true if the values are shard keys.
+	var sliceFirstColumn = func(values [][]interface{}, sk bool) (ret []string) {
+		ret = make([]string, 0, len(values))
+		if sk {
+			for i := range values {
+				ret = append(ret, values[i][0].([]string)...)
```
Comment (xuthus5) [inline lib/util/lifted/influx/coordinator/statement_executor.go:1438]: give bounds checking in place, how to do if `values[i][0]` is nil? if the interface assertion does not meet the expectation?

File: lib/util/lifted/influx/coordinator/statement_executor.go [hunk 2]
```diff
@@ -1402,6 +1407,130 @@ func (e *StatementExecutor) executeShowMeasurementsStatement(q *influxql.ShowMea
...
+			}
+		}
+		sort.Strings(ret)
+		return
+	}
+	var emitted = false
+	for key, m := range measurements {
+		originName := m.OriginName()
+		row := &models.Row{Name: originName, Tags: make(map[string]string, 1), Columns: []string{"Detail"}}
+		values := make([][]interface{}, 0, 8)
```
Comment (xuthus5) [inline lib/util/lifted/influx/coordinator/statement_executor.go:1452]: confusing magic number `8`, can you give me a brief description?

**Comment by xiangyu5632** [comment]
> \> show measurements detail with measurement = mst
name: mst
tags: Retention policy=rp0

Can the `tags` be used instead of the `retention policy`?

File: lib/util/lifted/influx/coordinator/statement_executor.go (hunk 2 above)
Comment (StepY1aoZz) [reply lib/util/lifted/influx/coordinator/statement_executor.go:1452]: The `values` here, which represents the detailed schema information of measurements, has 8 rows at most, including shard keys, indexes, engine type, sort keys, primary keys, compaction type, field keys and tag keys.

**Comment by StepY1aoZz** [comment]
> > > show measurements detail with measurement = mst
> > name: mst
> > tags: Retention policy=rp0
> 
> Can the `tags` be used instead of the `retention policy`?

For now, the `retention policy` info is stored in `tags` property of `models.Rows`, and the `tags` part is rendered in `ts-cli`. Thus, I think it is not easy to replace it if we still want the policy info to be located outside the `Detail` table.
However, we can put the policy info into the table, which looks like this:
```
> show measurements detail with measurement = mst
name: mst
+--------------------------------+
|             Detail             |
+--------------------------------+
| retention policy: rp0          |
| INDEX: <nil>                   |
| SHARD KEY: <nil>               |
| ENGINE TYPE: tsstore           |
| TAG KEYS: country, name        |
| FIELD KEYS: address(string),   |
| age(float), alive(boolean),    |
| height(integer)                |
+--------------------------------+
1 columns, 5 rows in set

``` 
Is that looks better for you?

**Comment by xiangyu5632** [comment]
> > > > show measurements detail with measurement = mst
> > > > name: mst
> > > > tags: Retention policy=rp0
> > 
> > 
> > Can the `tags` be used instead of the `retention policy`?
> 
> For now, the `retention policy` info is stored in `tags` property of `models.Rows`, and the `tags` part is rendered in `ts-cli`. Thus, I think it is not easy to replace it if we still want the policy info to be located outside the `Detail` table. However, we can put the policy info into the table, which looks like this:
> 
> ```
> > show measurements detail with measurement = mst
> name: mst
> +--------------------------------+
> |             Detail             |
> +--------------------------------+
> | retention policy: rp0          |
> | INDEX: <nil>                   |
> | SHARD KEY: <nil>               |
> | ENGINE TYPE: tsstore           |
> | TAG KEYS: country, name        |
> | FIELD KEYS: address(string),   |
> | age(float), alive(boolean),    |
> | height(integer)                |
> +--------------------------------+
> 1 columns, 5 rows in set
> ```
> 
> Is that looks better for you?

better.
retention policy --> RETENTION POLICY

File: lib/util/lifted/influx/coordinator/statement_executor.go (hunk 1 above)
Comment (StepY1aoZz) [reply lib/util/lifted/influx/coordinator/statement_executor.go]: Fixed. Thanks for the suggestion.

**Comment by StepY1aoZz** [comment]
@xiangyu5632 Fixed. Now the command returns as following:

```
> CREATE MEASUREMENT db0.rp0.cpu (region tag, az tag, v1 int64, v2 float64, v3 bool, v4 string) WITH  ENGINETYPE = columnstore  SHARDKEY az,region PRIMARYKEY az,region,time
> show measurements detail with measurement = cpu
name: cpu
+--------------------------------+
|             Detail             |
+--------------------------------+
| RETENTION POLICY: rp0          |
| INDEX: <nil>                   |
| SHARD KEY: az, region          |
| ENGINE TYPE: columnstore       |
| PRIMARY KEY: az, region, time  |
| SORT KEY: az, region, time     |
| COMPACTION_TYPE: row           |
| TAG KEYS: az, region           |
| FIELD KEYS: v1(integer),       |
| v2(float), v3(boolean),        |
| v4(string)                     |
+--------------------------------+
1 columns, 9 rows in set

```

File: lib/util/lifted/influx/coordinator/statement_executor.go (hunk 2 above)
Comment (StepY1aoZz) [reply lib/util/lifted/influx/coordinator/statement_executor.go]: After the policy information was added, now it becomes 9. I also leave a code comment here to improve its readability.

**Review by xiangyu5632 — APPROVED** [review]

**Comment by xiangyu5632** [comment]
@StepY1aoZz  please resolve the conflicting file： lib/util/lifted/influx/influxql/y.go

**Comment by StepY1aoZz** [comment]
I'm on vacation now and will return for this conflict on next week. Sorry for that.

**Comment by StepY1aoZz** [comment]
@xiangyu5632 The base of this commit has been updated to the latest and conflict has been resolved. Could you please check it out?

**Review by xiangyu5632 — APPROVED** [review]

**Pull request merged**